keras/src/callbacks/orbax_checkpoint.py

```diff
@@ -0,0 +1,497 @@
+import warnings
+
+import numpy as np
+
+from keras.src import backend
+from keras.src import tree
+from keras.src.api_export import keras_export
+from keras.src.callbacks.monitor_callback import (
+    MonitorCallback,  # For metric monitoring logic
+)
+from keras.src.utils.io_utils import print_msg
+from keras.src.utils.module_utils import ocp
+
+# Context and AsyncOptions are accessed through the lazy-loaded ocp module
+
+
+def _get_state_tree(model):
+    """Get the complete model state as a nested tree structure."""
+    # For JAX backend, preserve native arrays if JAX monitoring available
+    # to avoid unnecessary conversions. Otherwise convert to numpy.
+    if backend.backend() == "jax":
+        try:
+            import jax
+
+            # Check if jax.monitoring.record_scalar exists (JAX 0.7.0+)
+            jax.monitoring.record_scalar
```

Comment:
I'm confused by this. `jax.monitoring.record_scalar` is unrelated to this.

1. What is the issue with JAX < 0.7.0?
2. Why don't you check the jax version directly? https://github.com/keras-team/keras-rs/blob/5d7f18ae1103cec9a36fa621aa3f1f5d1128e2a7/keras_rs/src/layers/embedding/jax/distributed_embedding_test.py#L50